### PR TITLE
[MIG][13.0] account_invoice_margin: Migration to v13.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT21603

### CODE REVIEW

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @sergio-teruel. Thanks a lot for porting this module.

regarding the port from 12.0 to 13.0 : Code review / no test. :+1: Thanks.

regarding this module, I have a recurring problem with the design of this module, as it bases the computation of the margin of the invoice on all the invoice lines. In fact, some product (and so some lines) should not impact the margin computation. A typical exemple is for "Down Payment"

**Step to reproduce :**
- create a sale order with saleable product.
- create an invoice (A) with a down Payment.
- Create an invoice (B) with the rest of the order to invoice.

image


**Current result**
the margin of the invoice B has an incorrect margin, specially the percent value.

**Desired result**
- the margin of the invoice B should be based on the product line, and not on the Down Payment line

**What I propose**
- add a boolean ``impact_margin`` on the ``product.template`` / ``product.product`` models, (``default=True``)
- filter the computation of the margin on invoice lines with product with ``impact_margin`` to True.
- Do not display margin and margin_percent on invoice line for product with ``impact_margin`` to False.

1) are you agree with the limitation of this module, regarding Down Payment ? (and certainly other kind of products)
2) If yes, are you agree for the design I propose ?
3) If yes, I can make a PR against your PR, if you want.

kind regards.

File: account_invoice_margin/security/account_invoice_margin_security.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 Sergio Teruel - Tecnativa <sergio.teruel@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<openerp>
```
Comment (manuelcalerosolis) [inline account_invoice_margin/security/account_invoice_margin_security.xml]: <odoo>

**Review by manuelcalerosolis — APPROVED** [review]

File: account_invoice_margin/security/account_invoice_margin_security.xml (hunk 1 above)
Comment (manuelcalerosolis) [reply account_invoice_margin/security/account_invoice_margin_security.xml]: Code and functional review.

**Comment by pedrobaeza** [comment]
@sergio-teruel please check @legalsylvain comments and replace some obsolete `<openerp>` tags.

**Comment by sergio-teruel** [comment]
@legalsylvain Hi, you are right with the margin behavior for invoices with deposit payments...
mybe we could excluded down payment product for margin computation...How about it?

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done

**Comment by sergio-teruel** [comment]
@legalsylvain About your proposal, I agree.. so if you prefer this option perfect... I am thinking that theese alow to change down payment product ... 
You can make a PR to my PR.... :smile:

**Comment by sergio-teruel** [comment]
@legalsylvain Can you review last commit to exclude downpayment lines ??

File: account_invoice_margin/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (Tardo) [inline account_invoice_margin/models/__init__.py:1]:
```suggestion
# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
```

File: account_invoice_margin/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
+# © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (Tardo) [inline account_invoice_margin/models/account_invoice.py]:
```suggestion
# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
```

**Review by Tardo — APPROVED** [review]
:+1: Code review, only non-blocking comments

**Comment by sergio-teruel** [comment]
@Tardo Changes done

**Comment by pedrobaeza** [comment]
@legalsylvain everything OK in your part?

**Comment by sergio-teruel** [comment]
Anybody can merge??

**Pull request merged**